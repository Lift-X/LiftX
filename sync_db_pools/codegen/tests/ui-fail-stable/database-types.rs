../ui-fail/database-types.rs