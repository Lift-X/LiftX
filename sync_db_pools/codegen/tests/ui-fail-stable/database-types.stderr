error[E0277]: the trait bound `Unknown: Poolable` is not satisfied
  --> tests/ui-fail-stable/database-types.rs:6:10
   |
6  | struct A(Unknown);
   |          ^^^^^^^ the trait `Poolable` is not implemented for `Unknown`
   |
note: required by a bound in `rocket_sync_db_pools::Connection`
  --> $WORKSPACE/contrib/sync_db_pools/lib/src/connection.rs
   |
   | pub struct Connection<K, C: Poolable> {
   |                             ^^^^^^^^ required by this bound in `rocket_sync_db_pools::Connection`

error[E0277]: the trait bound `Unknown: Poolable` is not satisfied
  --> tests/ui-fail-stable/database-types.rs:5:1
   |
5  | #[database("foo")]
   | ^^^^^^^^^^^^^^^^^^ the trait `Poolable` is not implemented for `Unknown`
   |
note: required by a bound in `ConnectionPool`
  --> $WORKSPACE/contrib/sync_db_pools/lib/src/connection.rs
   |
   | pub struct ConnectionPool<K, C: Poolable> {
   |                                 ^^^^^^^^ required by this bound in `ConnectionPool`
   = note: this error originates in the attribute macro `database` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Vec<i32>: Poolable` is not satisfied
  --> tests/ui-fail-stable/database-types.rs:9:10
   |
9  | struct B(Vec<i32>);
   |          ^^^ the trait `Poolable` is not implemented for `Vec<i32>`
   |
note: required by a bound in `rocket_sync_db_pools::Connection`
  --> $WORKSPACE/contrib/sync_db_pools/lib/src/connection.rs
   |
   | pub struct Connection<K, C: Poolable> {
   |                             ^^^^^^^^ required by this bound in `rocket_sync_db_pools::Connection`

error[E0277]: the trait bound `Vec<i32>: Poolable` is not satisfied
  --> tests/ui-fail-stable/database-types.rs:8:1
   |
8  | #[database("foo")]
   | ^^^^^^^^^^^^^^^^^^ the trait `Poolable` is not implemented for `Vec<i32>`
   |
note: required by a bound in `ConnectionPool`
  --> $WORKSPACE/contrib/sync_db_pools/lib/src/connection.rs
   |
   | pub struct ConnectionPool<K, C: Poolable> {
   |                                 ^^^^^^^^ required by this bound in `ConnectionPool`
   = note: this error originates in the attribute macro `database` (in Nightly builds, run with -Z macro-backtrace for more info)
